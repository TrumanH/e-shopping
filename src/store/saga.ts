import { categoriesSaga } from './categories/categories.saga';
import createSagaMiddleware from 'redux-saga';
import { all, call } from 'redux-saga/effects';

export function* rootSaga() {
    yield all([call(categoriesSaga)]);
  }

// create the saga middleware
export const sagaMiddleware = createSagaMiddleware();