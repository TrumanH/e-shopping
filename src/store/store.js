import { configureStore } from '@reduxjs/toolkit';
import userReducer from './user/user.slice';
import categoriesReducer from './categories/categories.slice';
import cartReducer from './cart/cart.slice';
// root reducer

export const store = configureStore({
  reducer: {
    user: userReducer,
    categories: categoriesReducer,
    cart: cartReducer,
  },
});