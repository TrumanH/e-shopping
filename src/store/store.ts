import userReducer from './user/user.slice';
import categoriesReducer from './categories/categories.slice';
import cartReducer from './cart/cart.slice';
import { configureStore, combineReducers } from '@reduxjs/toolkit';
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import { sagaMiddleware, rootSaga } from './saga';

const rootReducer = combineReducers({
  user: userReducer,
  categories: categoriesReducer,
  cart: cartReducer,
});

const persistConfig = {
  key: 'root',
  version: 1,
  storage,
  // blacklist: ['user'],
  whitelist: ['cart'],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        // Ignore these action types
        ignoredActions: ['user/setUser', FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
        // Ignore these field paths in all actions
        ignoredActionPaths: ['meta.arg', 'payload.timestamp'],
        // Ignore these paths in the state
        ignoredPaths: ['user.user'],
      },
    }).concat(sagaMiddleware),
  devTools: process.env.NODE_ENV !== 'production', // The Redux DevTools Extension is disabled for production
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;

// then run the saga
sagaMiddleware.run(rootSaga)

export const persistor = persistStore(store);

// https://redux-toolkit.js.org/usage/usage-guide#use-with-redux-persist
