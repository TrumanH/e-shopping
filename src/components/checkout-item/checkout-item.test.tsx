
import CheckoutItem from './checkout-item.component';
import {render, screen} from "@testing-library/react";
import { BrowserRouter } from 'react-router-dom';
import {Provider} from "react-redux";
import {store} from "../../store/store";

describe('CheckoutItem component', () => {
    const item = {
        id: "hat1",
        name:"Brown Brim", 
        imageUrl:"https://i.ibb.co/ZYW3VTp/brown-brim.png", 
        price: 25, 
        quantity:2
    }
    const produceComponent = () => {
        return render(
          <BrowserRouter>
            <Provider store={store}>
              <CheckoutItem cartItem={item} />
            </Provider>
          </BrowserRouter>
          
        );
    }
    test('match the snapshot', ()=>{
        const { asFragment } = produceComponent();
        expect(asFragment()).toMatchSnapshot();
    });
    test('render the CheckoutItem component, get item name and price', ()=>{
        produceComponent();
        expect(screen.getByText(/Brown Brim/i)).toBeInTheDocument();
        expect(screen.getByText(/25/i)).toBeInTheDocument();
    });
});