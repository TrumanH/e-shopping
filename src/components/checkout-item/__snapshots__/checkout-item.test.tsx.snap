// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem component match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe geWseC"
  >
    <div
      class="sc-gswNZR bMaIyz"
    >
      <img
        alt="Brown Brim"
        src="https://i.ibb.co/ZYW3VTp/brown-brim.png"
      />
    </div>
    <span
      class="sc-dkrFOg hlmRre"
    >
      Brown Brim
    </span>
    <span
      class="sc-dkrFOg sc-hLBbgP hlmRre gJsMjZ"
    >
      <div
        class="sc-eDvSVe gTUNTl"
      >
        ❮
      </div>
      <span
        class="sc-jSUZER blfUCp"
      >
        2
      </span>
      <div
        class="sc-eDvSVe gTUNTl"
      >
        ❯
      </div>
    </span>
    <span
      class="sc-dkrFOg hlmRre"
    >
       25
    </span>
    <div
      class="sc-gKPRtg jRpdXo"
    >
      ✕
    </div>
  </div>
</DocumentFragment>
`;
